### DISCUSSION TITLE
Server sustainability

### DISCUSSION THREAD

**Discussion by heyjaywilson** [iOSDevSpace/iosdevspace#6] [General]
# Server sustainability

Server sustainability is a big thing to think about. No one wants their instance to just go poof one day, and I most definitely do not.

### How is hosting handled currently?

It’s slightly outlined in the [readme](https://github.com/heyjaywilson/iosdevspace) under [Server Information](https://github.com/heyjaywilson/iosdevspace#server-information). I use Masto.host to host the instance, and I purchased the domain through hover.com. Utilizing Masto.host allows us to really scale and not worry too much about maintenance and such. I think that's a big win and allows us to really focus on keeping the instance a pleasant place to be. Also leaning on Masto.host means we can usually "throw money at the problem" of scaling and not have to deal with the technical sides of things.

#### Cons to this:

- Masto.host could shut down at some point
- We could eventually exceed the recommended amount of people no matter how much we pay

#### Pros to this:

- No one has to be a linux expert
- No one has to worry about security or updates

### Thoughts around hosting sustainability

Hosting should really stay with Masto.host. I am not an expert in server side maintenance nor do I want to be.

I am looking into self-hosting so we can scale or save money as needed, but right now at the size we are it’s about a wash for everything.

### How is hosting paid for?

I paid for the initial plan and domain. The rest has come from supporters/donations on the [ko-fi](https://ko-fi.com/iosdevspace) page, which is awesome! Some people have chosen to become a member+ and donate $1/month and others have donated larger amounts on a one time basis. The first being CocoheadsNL which allowed us to make a huge leap forward into the next level of hosting.

I have two membership levels setup currently, one for member+ ($1/month) and one for iOS Dev Blue ($8/month) and then people can make one time donations.

Just for info: we have about 5 months worth of the next level of hosting if/when we upgrade

#### Cons to this:

- People could stop supporting the instance

#### Pros to this:

- One person doesn’t really “own” the server

### Thoughts around monetary sustainability

We could become a paid for instance, meaning you must make a donation to join

- [Android Dev instance](https://androiddev.social/about) does this
   - I’m not the biggest fan cause I think something like this instance should be open/welcoming to all

We could just stay as we are with the donation based model

- [Fosstodon](https://hub.fosstodon.org/support/) does something similar
- In addition to staying as we are, I can add roles for members and one time supporters so that they get a badge on their profile.

## Moderation/Administration

Currently it's just @kylereddoch and I as mods and admin. Besides approving people, there's not much more to it. I'm sure as we grow there could be and I might want to look for another person or 2. This doesn't really worry me too much right now.

**Comment by babbage** [comment]
This is a really useful discussion for us to have well before there is a crisis. @heyjaywilson you've done a great job bringing this community together and are doing a great job supporting it, with @kylereddoch too. Thanks both!

I 100% support the idea that iosdev.space continue to use managed hosting like Masto.host. Makes so much sense to me that we outsource the server maintenance to people who are fully focussed on this, even if it costed a bit more, and unless it got truly prohibitively expensive.

I'm quite surprised at how relatively inexpensive it would be to scale to 2,000 users, on a per user/per year cost basis (currently US$89/month, or just under 50c per user per year). Maybe it really would be sustainable to support the server with a relatively small subset of people who chose to donate. But that would presume that you continue to get a non-trivial percentage of people to donate enough to cover these growing costs as the server scales.

If there was going to be an expectation of support from all users*, one thought I had is that there could be tiers that users could self-select into based on country/region of origin (to balance out purchasing power parity) and/or life stage (student/unwaged vs. employed or similar). e.g. paying US$1/month in countries with a higher earning power, while offering $1/year to people where earnings are relatively lower and then a one-off donation of $1 for still others—which could be waived by an admin? 

\* I am not at this point wanting to advocate that this is the right course. Just discussing what the possible options might be.

**Reply by heyjaywilson** [reply]
+100000 for keeping iosdev.space on a managed host. It’s just a lot of worry off of my back. Usually, managed hosts keep close to the pricing of a self hosted option that the extra is worth it.

I like the tier’s ideas. It’s also part of the reason I went with ko-fi for donations. It easily allows tiers and 1 time donations.

**Comment by jeroenleenarts** [comment]
I think having the runway of 5 months is good. I understood there is a big difference between total users and active users.

One thing you could look into is in working towards sponsorships from companies. Give a corp sponsor a specific perk, shoutouts, banner placement, etc. It could be useful if the money streaming through the server becomes "bigger" to move into some legal entity. Open Collective provides a very low barrier of entry to get under the umbrella of a legal entity and will put corp sponsors at ease for not wanting to provide money to an individual without oversight or potential tax issues.

**Reply by heyjaywilson** [reply]
The runway is great! I also want to see the difference between active/total users before upgrading... I just don't know if people will "forget" about it or not use it as much. It's definitely getting popular so the active vs total number might get closer.

I am treating this as an extension of my LLC, CCT Plus. My LLC is setup to support all my social networking and iOS business. But if this does grow or the income/donations become way more I think setting something up separately for it makes sense.

I definitely think corporate sponsorship could be an interesting route to take and something to think about. I'll also look into Open Collective to just see how that works.

**Reply by heyjaywilson** [reply]
I don't think I have it anywhere, but currently the donations go into a Paypal account that is only for iOS Dev Space and Masto.Host is paid for from that paypal account.

~~Currently looking into Open Collective though so I can see what it would take. I like the openness/transparency from it.~~ I opened an [Open Collective](https://opencollective.com/iosdevspace).
